src/main/java/org/opensearch/security/configuration/ConfigurationRepository.java

```diff
@@ -609,36 +607,47 @@ public Map<CType, SecurityDynamicConfiguration<?>> getConfigurationsFromIndex(
                 } else {
                     LOGGER.debug("security index exists and was created with ES 7 (new layout)");
                 }
-                retVal.putAll(
-                    validate(cl.load(configTypes.toArray(new CType[0]), 10, TimeUnit.SECONDS, acceptInvalid), configTypes.size())
+                resultBuilder.with(
+                    validate(cl.load(configTypes.toArray(new CType<?>[0]), 10, TimeUnit.SECONDS, acceptInvalid), configTypes.size())
                 );
 
             } else {
                 // wait (and use new layout)
                 LOGGER.debug("security index not exists (yet)");
-                retVal.putAll(
-                    validate(cl.load(configTypes.toArray(new CType[0]), 10, TimeUnit.SECONDS, acceptInvalid), configTypes.size())
+                resultBuilder.with(
+                    validate(cl.load(configTypes.toArray(new CType<?>[0]), 10, TimeUnit.SECONDS, acceptInvalid), configTypes.size())
                 );
             }
 
         } catch (Exception e) {
             throw new OpenSearchException(e);
         }
 
+        ConfigurationMap result = resultBuilder.build();
+
         if (logComplianceEvent && auditLog.getComplianceConfig() != null && auditLog.getComplianceConfig().isEnabled()) {
-            CType configurationType = configTypes.iterator().next();
+            CType<?> configurationType = configTypes.iterator().next();
             Map<String, String> fields = new HashMap<String, String>();
-            fields.put(configurationType.toLCString(), Strings.toString(MediaTypeRegistry.JSON, retVal.get(configurationType)));
+            fields.put(configurationType.toLCString(), Strings.toString(MediaTypeRegistry.JSON, result.get(configurationType)));
             auditLog.logDocumentRead(this.securityIndex, configurationType.toLCString(), null, fields);
         }
 
-        return retVal;
+        return result;
+    }
+
+    public SecurityDynamicConfiguration<?> getUnconvertedConfigurationFromIndex(CType<?> configType, boolean logComplianceEvent) {
+        ConfigurationMap map = this.getConfigurationsFromIndex(List.of(configType), logComplianceEvent, this.acceptInvalid);
+        SecurityDynamicConfiguration<?> result = map.get(configType);
+        if (result != null && result.getAutoConvertedFrom() != null) {
+            result = result.getAutoConvertedFrom();
+        }
+
+        return result;
     }
 
-    private Map<CType, SecurityDynamicConfiguration<?>> validate(Map<CType, SecurityDynamicConfiguration<?>> conf, int expectedSize)
-        throws InvalidConfigException {
+    private ConfigurationMap validate(ConfigurationMap conf, int expectedSize) throws InvalidConfigException {
 
-        if (conf == null || conf.size() != expectedSize) {
+        if (conf == null || conf.size() < expectedSize) {
```

Comment:
Same comment from other PR, is the operator change necessary? Can it ever be the case that `conf.size() > expectedSize`?

Reply:
Got it, the auto-conversion occurs during Config loading so the output could be n+1 the input if tenants config is generated. Thanks!